lxd/container_new.go

```diff
@@ -76,41 +79,79 @@ func getIps(c *lxc.Container) []shared.Ip {
 	return ips
 }
 
-func newStatus(c *lxc.Container, state lxc.State) shared.ContainerStatus {
-	status := shared.ContainerStatus{State: state.String(), StateCode: shared.State(int(state))}
-	if state == lxc.RUNNING {
-		status.Init = c.InitPid()
-		status.Ips = getIps(c)
-	}
-	return status
+type container interface {
+	RenderState() (*shared.ContainerState, error)
+	Reboot() error
+	Freeze() error
+	IsPrivileged() bool
+	IsRunning() bool
+	IsEmpheral() bool
+	IsSnapshot() bool
+	Shutdown(timeout time.Duration) error
+	Start() error
+	Stop() error
+	Unfreeze() error
+	Delete() error
+	CreateFromImage(hash string) error
+	Restore(sourceContainer container) error
+	Copy(source container) error
+	Rename(newName string) error
+	IDGet() int
+	NameGet() string
+	ArchitectureGet() int
+	PathGet(newName string) string
+	RootfsPathGet() string
+	TemplatesPathGet() string
+	StateDirGet() string
+	TemplateApply(trigger string) error
+	LogFilePathGet() string
+	LogPathGet() string
+	InitPidGet() (int, error)
+	IdmapSetGet() (*shared.IdmapSet, error)
+	ConfigReplace(newConfig containerLXDArgs) error
+	ConfigGet() containerLXDArgs
```

Comment:
Can we sort those? would make reading a bit easier :)

I'm also not sure about all of those WhateverGet functions when there's no matching Set function.
The new design also makes it slightly harder and possibly slower to get to things like the container config which we previously had directly against the container.